src/Lean/Widget/UserWidget.lean

```diff
@@ -212,10 +213,12 @@ def erasePanelWidget [Monad m] [MonadEnv m] (h : UInt64) : m Unit := do
 /-- Save the data of a panel widget which will be displayed whenever the text cursor is on `stx`.
 `hash` must be `hash (toModule c).javascript`
 where `c` is some global constant annotated with `@[widget_module]`. -/
-def savePanelWidgetInfo [Monad m] [MonadEnv m] [MonadError m] [MonadInfoTree m]
-    (hash : UInt64) (props : StateM Server.RpcObjectStore Json) (stx : Syntax) : m Unit := do
+def savePanelWidgetInfo (hash : UInt64) (props : StateM Server.RpcObjectStore Json) (stx : Syntax) :
+    CoreM Unit := do
```

Comment:
@Vtec234 Do you see any downside in specializing this (and others?) to `CoreM`? I feel like we sometimes went overboard with monad generalization.

Reply:
I don't immediately see any downsides.